src/GeneratedHydrator/Configuration.php

```diff
@@ -220,4 +225,20 @@ public function getClassNameInflector()
 
         return $this->classNameInflector;
     }
+
+    /**
+     * @param \PhpParser\NodeVisitor[] $visitors
+     */
+    public function setCustomVisitors(array $customVisitors)
```

Comment:
Splatting isn't available before PHP 5.6 and the minimum required version for this library is now PHP 5.4. But I could do a check that all array elements are NodeVisitor instances.

As for constructor injection, custom visitors are optional so I don't think these belong in the constructor. IMHO only required settings should go in the constructor.